aptos-move/aptos-aggregator/src/aggregator_extension.rs

```diff
@@ -324,201 +309,120 @@ impl Aggregator {
     }
 
     /// Implements logic for adding to an aggregator.
+    /// Returns error if transaction is in invalid state, and should be re-executed.
+    /// Returns true if addition succeeded, and false if it would overflow.
     pub fn try_add(
         &mut self,
         resolver: &dyn AggregatorResolver,
         input: u128,
-    ) -> PartialVMResult<()> {
```

Comment:
we need to distinguish try_add needing to return false, from any other error - which we should propagate and abort the transaction

Reply:
So, we are returning Ok(false) when we know that re-execution the transaction again would also fail. We are returning Err(..) when the current execution failed, but re-execution could possibly succeed. Is that right?

Reply:
we are returning Ok(false) when try_add call should return false - i.e. called +1 when value=limit.  No re-execution needed (unless start_value changes)

we return Err(..) when try_add detected irreconcilable state - meaning that current execution is not valid - even if start_value doesn't change.